### PR TITLE
Fix empty str defaults in config/base.yml

### LINKED ISSUE #74580: Support empty string defaults for configuration settings
### Summary

As of https://github.com/ansible/ansible/pull/74523, using empty string defaults for config options in `lib/ansible/config/base.yml` causes the error: `WARNING: Inline literal start-string without end-string.`

The two configuration settings defaulting to empty strings were replaced with ~ (YAML equivalent to Python's None) for the time being and in this case it made no functional difference. Since the configuration settings are used both for documentation and in the code (and None != ''), empty string defaults should still be valid.

On a semi-related note, the file contains `default: null`, `default: ~`, ~`default:`~ (removed these in #74607), and settings that omit the default field, all of which are functionally equivalent but not displayed the same way.

### Issue Type

Documentation Report

### Component Name

docs/templates/config.rst.j2

### Ansible Version

```console
$ ansible --version
ansible [core 2.12.0.dev0]
```

### Configuration

```console
$ ansible-config dump --only-changed
N/A
```

### OS / Environment

N/A

### Additional Information

N/A

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by s-hertel**
##### SUMMARY
Fixes https://github.com/ansible/ansible/issues/74580

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
docs/templates/config.rst.j2

### CODE REVIEW

**Comment by s-hertel** [comment]
Re-verified

File: docs/templates/config.rst.j2 [hunk 1]
```diff
@@ -126,7 +126,9 @@ you can use the command line utility mentioned above (`ansible-config`) to brows
 {%     if config['type'] %}
 :Type: {{config['type']}}
 {%     endif %}
-{%     if 'default' in config %}
+{%     if 'default' in config and config['default'] == '' %}
+:Default: {{config['default']}}
+{%     elif 'default' in config %}
```
Comment (bcoca) [inline docs/templates/config.rst.j2]:
```suggestion
{% if 'default' in config %}
  {%- if config['default'] == '' %}
:Default: {{config['default']}}
  {%- else %}
```

Comment (s-hertel) [reply docs/templates/config.rst.j2]: Oh, thanks :smile:

**Comment by s-hertel** [comment]
I canceled the run before letting it finish but this is the failing sanity test:

```
01:10 Running sanity test "docs-build"
01:10 Run command: /root/.ansible/test/venv/sanity.docs-build/3.10/4d056497/bin/python /root/ansible/test/sanity/code-smell/docs-build.py
02:31 ERROR: Found 3 docs-build issue(s) which need to be resolved:
02:31 ERROR: docs/docsite/rst/index.rst:0:0: unknown: 
02:31 ERROR: docs/docsite/rst/index.rst:0:0: unknown: ````
02:31 ERROR: docs/docsite/rst/reference_appendices/config.rst:1082:0: warning: Unexpected section title or transition.
02:31 See documentation for help: https://docs.ansible.com/ansible-core/devel/dev_guide/testing/sanity/docs-build.html
02:31 Running sanity test "package-data"
02:31 Run command with data: /root/.ansible/test/venv/sanity.package-data/3.10/b269f971/bin/python /root/ansible/test/sanity/code-smell/package-data.py
02:46 FATAL: The 1 sanity test(s) listed below (out of 4) failed. See error output above for details.
02:46 docs-build
02:46 Run command with stdout: docker exec -i ansible-test-controller-nnDC6X9m sh -c 'tar cf - -C /root/ansible/test --exclude .tmp results | gzip'
02:46 Run command with stdin: tar oxzf - -C /__w/1/ansible/test
02:46 FATAL: Command "docker exec ansible-test-controller-nnDC6X9m /usr/bin/env ANSIBLE_TEST_CONTENT_ROOT=/root/ansible LC_ALL=en_US.UTF-8 /usr/bin/python3.10 /root/ansible/bin/ansible-test sanity -v --containers '{}' --junit --coverage-check --keep-git --test ansible-doc --test docs-build --test package-data --test changelog --allow-disabled --truncate 0 --color yes --host-path test/results/.tmp/host-d4ybyv3h --metadata test/results/.tmp/metadata-qb9gqwq9.json --base-branch origin/devel" returned exit status 1.
02:46 Run command: docker rm -f ansible-test-controller-nnDC6X9m
##[error]Bash exited with code '1'.
Finishing: Run Tests
```

File: docs/templates/config.rst.j2 (hunk 1 above)
Comment (bcoca) [reply docs/templates/config.rst.j2]: just my OCD against the same condition tested x2 ....

**Comment by samccann** [comment]
Hi @s-hertel - I can't tell if this should be backported to stable-2.14 or not??

**Pull request merged**